src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -227,6 +226,7 @@ export class HttpConnection implements IConnection {
         // as part of negotiating
         let url = this.baseUrl;
         this._accessTokenFactory = this._options.accessTokenFactory;
+        this._httpClient.replaceAccessTokenFactory(this._accessTokenFactory);
```

Comment:
Do we let anyone else reference the `AccessTokenHttpClient` we create? How would of the `_accessTokenFactory` changed from when we constructed it?

Reply:
> Do we let anyone else reference the AccessTokenHttpClient we create?

No.

> How would of the _accessTokenFactory changed from when we constructed it?

Negotiate redirects give a new token factory, and when the client restarts after a negotiate redirect we want the original factory back.